inrupt/pod-browser
# Fix/requesting dataset twice

This PR fixes bug where every resource was requested twice. When navigating to a container, two GET requests were made for the container.

The bug was caused by a hook that fetched the IRIs for the resources within a container. Instead of using the dataset directly, it was fetching the dataset again, and then fetching the IRIs. 

Since we already do a GET request for the dataset itself before fetching the IRIs for the resources, it made more sense to use the dataset directly. In addition, we also used the container Thing to get the IRIs, so I created a container model and we now use the model in the Container view component and the function that fetches the IRIs. 

**Update**:

After looking into it, I could not get the `mutate` function from `useSWR` to behave in the way it is supposed to behave (or the way I would expect it to behave). The only way to get `mutate` to update the data properly was to pass it the newly fetched data, which kind of beats the purpose since what I wanted was to remove the extra request for the dataset. I decided to rewrite the hook to simply use a `useEffect`. 

I also introduced a spinner when moving from one folder to another since the data takes a bit longer than the breadcrumbs to update and it can be disconcerting. 

# To test:
- navigate to a container
- in the Network tab, verify that only one GET request is made when navigating to the container
